Rest/Controller/ContentController.php

```diff
@@ -73,13 +75,31 @@ public function __construct(
      *
      * @return \EzSystems\RecommendationBundle\Rest\Values\ContentData
      *
-     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundExceptionif the content, version with the given id and languages or content type does not exist
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException if the content, version with the given id and languages or content type does not exist
      * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException If the user has no access to read content and in case of un-published content: read versions
      */
     public function getContent($contentIdList)
     {
         $contentIds = explode(',', $contentIdList);
-        $content = $this->prepareContent($contentIds);
+        $lang = $this->request->get('lang');
+
+        $criterion = $contentTypesCriterion = array_map(function ($contentId) {
+            return new Criterion\ContentId($contentId);
+        }, $contentIds);
+
+        if (!$this->request->get('hidden')) {
+            $criterion[] = new Criterion\Visibility(Criterion\Visibility::VISIBLE);
+        }
+        if ($lang) {
+            $criterion[] = new Criterion\LanguageCode($lang);
+        }
+
+        $query = new Query();
+        $query->query = new Criterion\LogicalAnd($criterion);
+
+        $contents = $this->searchService->findContent($query)->searchHits;
```

Comment:
But on the other side you might get several locations for same content, so ignore this for now.
What we can do is refactor this later to use findContentInfo, which is added in upcoming 5.4.5 and Platform 1.0.